sys/ztimer/core.c

```diff
@@ -86,6 +86,12 @@ void ztimer_set(ztimer_clock_t *clock, ztimer_t *timer, uint32_t val)
     }
 
     timer->base.offset = val;
+
+    /* if this is the first timer for this clock, acquire it */
+    if (clock->ops->acquire && clock->list.next == NULL) {
+        clock->ops->acquire(clock);
+    }
+
```

Comment:
If I understand ztimer correctly, this is the best place to put this bit if we do not want to loose accuracy when setting up new timers. But in this case `clock->ops->now()` is called before `clock->ops->acquire()`. This is not harmful if we are just interacting with `pm_layered` but could become a problem in the future if we actually want to power on the timer using this callback.